### DISCUSSION TITLE
please include a way to insert audio file using open xml

### DISCUSSION THREAD

**Discussion by aman-gill22** [ShapeCrawler/ShapeCrawler#158] [Ideas]
brother i appreciate the hardwork u have done but i was expecting if u could include a way to insert audio files into ppt

**Comment by AdamShakhabov** [comment]
Hi. Thank you for your feedback :)

Audio? Why not. Will research this.

**Reply by aman-gill22** [reply]
thanks brother much appreciated

**Comment by AdamShakhabov** [comment]
@aman-gill22, a new **[v0.24.0](https://github.com/ShapeCrawler/ShapeCrawler/releases/tag/v0.24.0)** version has been released. Now you can add audio:

```C#
IPresentation presentation = SCPresentation.Open("helloWorld.pptx", isEditable: true);                
IShapeCollection shapes = presentation.Slides[0].Shapes;
using Stream mp3Stream = File.OpenRead("audio.mp3");
IAudioShape addedAudioShape = shapes.AddNewAudio(xPixel: 300, yPixels: 100, mp3Stream);

presentation.Save();
presentation.Close();
```

Let me know if it does not work correctly.

**Comment by aman-gill22** [comment]
Can't thank u enough brother....
I am working on an office automation project, once it is done will share the details with u ... meanwhile this new update will come in handy😁🙏
